plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/AdvancedConnectionEditor.java

```diff
@@ -276,6 +289,18 @@ protected GridLayoutFactory adjustAdvancedCompositeLayout(GridLayoutFactory grid
 		return composite;
 	}
 
+	private void discoverRegistryPressed(Shell shell) {
+		String ret = RegistryProviderModel.getDefault().getRegistryURL(pageModel.getSelectedConnection());
+		if( ret != null ) {
+			registryURLObservable.setValue(ret);
+		} else {
+			MessageBox messageBox = new MessageBox(shell, SWT.ICON_ERROR | SWT.OK);
+		    messageBox.setMessage("No registry provider found for the given connection.");
```

Comment:
In my case the user error is that I didnt have the adapter started (launched the cdk from cmd-line). So some hint like "You may have to start your CDK server adapter" would be very helpful to the dummy user (like me)